extension/src/popup/components/sendPayment/SendTo/index.tsx

```diff
@@ -220,29 +220,31 @@ export const SendTo = ({ previous }: { previous: ROUTES }) => {
                   <div className="SendTo__subheading">RECENT</div>
                 )}
                 <ul className="SendTo__recent-accts-ul">
-                  {recentAddresses.map((pubKey) => (
-                    <li key={pubKey}>
+                  {recentAddresses.map((address) => (
+                    <li key={address}>
                       <button
                         onClick={async () => {
                           setIsLoading(true);
-                          setValidatedPubKey(pubKey);
                           // recentAddresses already validated so safe to dispatch
-                          if (isFederationAddress(pubKey)) {
+                          if (isFederationAddress(address)) {
                             const fedResp = await FederationServer.resolve(
-                              pubKey,
+                              address,
                             );
-                            handleContinue(pubKey, fedResp.account_id);
+                            const publicKey = fedResp.account_id;
+                            setValidatedPubKey(publicKey);
+                            handleContinue(publicKey, address);
```

Comment:
we had a bug (on the old ln 234) where we sent the params in the wrong order: param 1 is supposed to be the public key, while param 2 is supposed to be the federated address. I think it was a little confusing because the var `pubKey` could either be a public key or a federated address. 

I renamed it to something a little more generic (`address`) so it was clear it could be public key *or* federated address. I created this var `publicKey` to make it clear `fedResp.account_id` is actually a public key